frontend/src/metabase/containers/DataPicker/useDataPickerValue.ts

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useState } from "react";
+
+import { DataPickerValue } from "./types";
+
+function cleanSchemaValue({ databaseId, schemaId }: Partial<DataPickerValue>) {
+  return databaseId ? schemaId : undefined;
+}
+
+function cleanTablesValue({
+  databaseId,
+  schemaId,
+  tableIds,
+}: Partial<DataPickerValue>) {
+  if (!tableIds) {
+    return [];
+  }
+  return databaseId && schemaId ? tableIds : [];
+}
+
+function cleanValue(value: Partial<DataPickerValue>): DataPickerValue {
+  return {
+    databaseId: value.databaseId,
+    schemaId: cleanSchemaValue(value),
+    tableIds: cleanTablesValue(value),
+  };
+}
+
+type HookResult = [DataPickerValue, (value: DataPickerValue) => void];
+
+function useDataPickerValue(
```

Comment:
maybe just `useDataPicker`? seems like a primary hook for the component. Also not clear what `value` means for the data picker

Reply:
Yeah, I can see it being a bit unobvious. I had plans for `useDataPicker`, though. One of the things we want to do next in data apps is to render the data search input outside of the picker (in a similar way to QB filtering modal search). I was going to introduce data picker context for that and was going to use `useDataPicker` to access context value

> Also not clear what `value` means for the data picker

Yeah, I was hoping the typings would make it easier to gasp. Do you happen to have ideas for a better name?